Added parser support for EK80 MRU1

praneethratna opened the pull request:
Addresses #1239 and now EK80 files of type MRU version 1 can be parsed.

CC @leewujung

leewujung commented on echopype/convert/utils/ek_raw_parsers.py:
```diff
@@ -522,7 +522,17 @@ def __init__(self):
                 ("heading", "f"),
-            ]
+            ],
+            1: [
```
Could you add in the docstring regarding MRU1 type in addition to MRU0?

praneethratna replied:
Updated!

leewujung reviewed (commented):
@praneethratna : Thanks for putting this in! I just have a minor comment about adding MRU1 in the docstring. I verified that the rest of `open_raw` runs through fine with the pitch, roll, heave data stored in the Platform group as pitch, roll, and _vertical_offset_.

But the parsed `heading` variable is not stored. Let me take a look at the convention to see if there's a variable we can store it, and if not, we can put it in the Platform group just with `heading` as the name. I will get back to you on this!

It'll be nice to have a test file in the CI for this.

@jmjech: Do you have a file that is smaller (say a few MB) that we can use as a test file? It adds to the time for tests to run through if the test files are large, so we usually try to include only small files.

Thanks both!

jmjech commented:
I e-mailed a 2 MB file to Wu-Jung. Let me know if it works for you. I have other "smaller" files.
For some reason it is not allowing me to add the data files here.

leewujung commented:
@jmjech: Thank you! The file works and does contain MRU1 datagrams.

@praneethratna: I have added this file (`20231016_Cal_-D20231016-T220322.raw`) under `test_data/ek80`. Could you please add a test for parsing it? In looking into this file I also checked for `np.all(echodata["Platform"]["pitch"].data == np.array(parser.mru["pitch"]))` and the same thing for `heave` (compare with `vertical_offset` in Platform) and `roll` (compare with `roll` in Platform). Would be good if we have these in the tests, by using the parser object https://github.com/OSOceanAcoustics/echopype/blob/03e3b52bb522222f5cdf887b6f519e47d4aae6bf/echopype/convert/api.py#L423

leewujung commented:
Also, on storing the `heading` variable. It is indeed in the ver.1 convention as below:

<img width="912" alt="Screenshot 2023-12-14 at 11 43 23 AM" src="https://github.com/OSOceanAcoustics/echopype/assets/15334215/ac2a2110-10f8-46c3-89be-9301f0fd2176">

@praneethratna: Could you please add this to the set group so that it gets stored together with the other variables? As you have seen, they are on the same timebase since they are all from the MRU* datagrams. Thanks!

praneethratna commented:
@leewujung I have added docstring for new `MRU1` datagram and also added a new test case using the file given by @jmjech. Also, now the `heading` variable will be stored in `Platform` group with the attributes which you've mentioned. Thanks!

leewujung reviewed (approved):
@praneethratna : Thanks for adding `heading` and test. I'll merge this now. :)
